app/services/Worker.php

```diff
@@ -3,70 +3,83 @@
 namespace GanttDashboard\App\Services;
 
 use GanttDashboard\App\Models\Workers;
-use Phalcon\Session\Adapter\Files as SessionAdapter;
-use Phalcon\Security;
+use GanttDashboard\App\Validators\Worker as WorkerValidator;
+use GanttDashboard\App\Services\Authentication as AuthenticationService;
+use Phalcon\Flash\Session as FlashSession;
+use Phalcon\Mvc\View;
 
-class Worker
+class Worker extends Base
 {
     /**
-     * @var SessionAdapter
+     * @var AuthenticationService
      */
-    private $sessionService;
+    private $authenticationService;
 
     /**
-     * @var Security
+     * @var WorkerValidator
      */
-    private $securityService;
+    private $workerValidator;
 
     /**
      * @var Workers
      */
     private $workerModel;
 
     /**
-     * Constructs the needed services, set in DI, for session, security and model
-     * @param SessionAdapter $sessionService
-     * @param Security $securityService
+     * Constructs the needed services, set in DI, for session, security, model,
+     * flashSession and view
+     * @param Authentication $authenticationService
+     * @param WorkerValidator $workerValidator
      * @param Workers $workerModel
+     * @param FlashSession $flashSession
+     * @param View $view
      */
     public function __construct(
-        SessionAdapter $sessionService,
-        Security $securityService,
-        Workers $workerModel
+        AuthenticationService $authenticationService,
+        WorkerValidator $workerValidator,
+        Workers $workerModel,
+        FlashSession $flashSession,
+        View $view
```

Comment:
why in the name of the god the view is injected into a service?
only the controller communicates with the view